File: tip5.js
function sumaNumerosPares(rango) {
    let resultado = 0;
    for (let numero = 1; numero <= rango; numero++) {
        if (numero % 2 === 0) {
            resultado += numero;
        }
    }
    return resultado;
}

const rango = 10;
const resultado = sumaNumerosPares(rango);
console.log(`La suma de números pares hasta ${rango} es: ${resultado}`);

function contarFrecuenciaPalabras(listaPalabras) {
    const frecuencia = {};

    for (const palabra of listaPalabras) {
        frecuencia[palabra] = (frecuencia[palabra] || 0) + 1;
    }

    return frecuencia;
}

const listaPalabras = ["gato", "perro", "gato", "ratón", "perro", "gato"];
const resultadoPalabras = contarFrecuenciaPalabras(listaPalabras);
console.log("Frecuencia de palabras:");
for (const palabra in resultadoPalabras) {
    console.log(`${palabra}: ${resultadoPalabras[palabra]}`);
}

File: tip5.py
def suma_numeros_pares(rango):
    resultado = 0
    for numero in range(1, rango + 1):
        if numero % 2 == 0:
            resultado += numero
    return resultado

rango = 10
resultado = suma_numeros_pares(rango)
print(f"La suma de números pares hasta {rango} es: {resultado}")

def contar_frecuencia_palabras(lista_palabras):
    frecuencia = {}
    for palabra in lista_palabras:
        if palabra in frecuencia:
            frecuencia[palabra] += 1
        else:
            frecuencia[palabra] = 1
    return frecuencia

lista_palabras = ["gato", "perro", "gato", "ratón", "perro", "gato"]
resultado = contar_frecuencia_palabras(lista_palabras)
print("Frecuencia de palabras:")
for palabra, conteo in resultado.items():
    print(f"{palabra}: {conteo}")
